### PR TITLE
#167484597 implemented like or dislike an article 

### PR DESCRIPTION
**Pull request by placideirandora**
#### What does this PR do?
It implements the approach of being able to like or dislike an article.

#### Description of Task to be done?
1. Adding routes for liking or disliking an article.
2. Adding controllers for liking or disliking an article.
3. Adding tests for liking or disliking an article.
4. Documenting the feature with Swagger.

#### Screenshots
Screen Shot 2019-08-21 at 23 13 31
Screen Shot 2019-08-21 at 23 13 38
Screen Shot 2019-08-21 at 17 21 57
Screen Shot 2019-08-21 at 23 13 22
Screen Shot 2019-08-21 at 23 11 37
Screen Shot 2019-08-21 at 23 11 56
Screen Shot 2019-08-21 at 23 12 06
Screen Shot 2019-08-21 at 23 12 26

#### How should this be manually tested?
1. Clone this branch.
2. Do **npm install**.
4. Do **npm start**
5. Create an article.
6. Like it with the following endpoint: **/api/v1/articles/{articleSlug}/like**
7. Dislike it with the following endpoint: **/api/v1/articles/{articleSlug}/dislike**
8. Retrieve the articles and check the total likes and dislikes.

### CODE REVIEW

**Review by raymond42 — COMMENTED** [review]
Good Job @placideirandora. I've tested these endpoints and they are working fine

**Review by raymond42 — APPROVED** [review]
Good Job @placideirandora. I've tested these endpoints and they are working fine

**Comment by placideirandora** [comment]
> Good Job @placideirandora. I've tested these endpoints and they are working fine

Thanks!

File: package.json [hunk 1]
```diff
@@ -12,7 +12,9 @@
     "test": "cross-env NODE_ENV=test nyc --reporter=text mocha  --require @babel/register --require @babel/polyfill  ./src/tests/*.js --timeout 30000 --exit",
     "down": "cross-env NODE_ENV=test node_modules/.bin/sequelize db:migrate:undo",
     "migrate": "cross-env NODE_ENV=test node_modules/.bin/sequelize db:migrate",
-    "seed": "node_modules/.bin/sequelize db:seed:all"
+    "seed": "node_modules/.bin/sequelize db:seed:all",
```
Comment (mkiterian) [inline package.json]: remove the additional scripts. Looks like @Malaba6 added these as well

File: src/controllers/articleController.js [hunk 2]
```diff
@@ -35,7 +37,7 @@ class ArticleController {
       payload.authorId = id;
       const article = await Article.create(payload);
       const {
-        slug, category, images, tagList, authorId
+        slug, category, images, tagList, authorId, likes, dislikes
```
Comment (mkiterian) [inline src/controllers/articleController.js]: I do not expect the payload when creating an article to have likes and dislikes, why are we getting them here?

File: src/controllers/articleController.js [hunk 3]
```diff
@@ -92,6 +96,266 @@ class ArticleController {
...
+    try {
+      const liker = req.userData.id;
+      const slugId = req.params.articleSlug;
+      const likeVote = 1;
+      const dislikeVote = 0;
+
+      const findArticle = await Article.findOne({ where: { slug: slugId } });
+
+      if (!findArticle) {
+        return res.status(200).json({
```
Comment (mkiterian) [inline src/controllers/articleController.js]: 200 is a success code, 404 would be much suitable here

File: src/controllers/articleController.js [hunk 4]
```diff
@@ -92,6 +96,266 @@ class ArticleController {
...
+          articleSlug, userId, likes
+        } = likedArticle;
+
+        if (likedArticle) {
+          await Article.increment({ likes: 1 }, { where: { slug: slugId } });
+
+          return res.status(200).json({
+            message: 'You have liked the article',
+            reaction: {
+              ArticleSLUG: articleSlug,
```
Comment (mkiterian) [inline src/controllers/articleController.js]: fix the capitalization here, use camelCase

**Review by mkiterian — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (placideirandora) [reply package.json]: Okay

File: src/controllers/articleController.js (hunk 2 above)
Comment (placideirandora) [reply src/controllers/articleController.js]: I wanted to give an article default likes and dislikes after being created. But, It's not necessary, I will display them only for the retrieval.

File: src/controllers/articleController.js (hunk 3 above)
Comment (placideirandora) [reply src/controllers/articleController.js]: You're Right. I missed that! I'm gonna change it.

File: package.json (hunk 1 above)
Comment (placideirandora) [reply package.json]: Excuse me... Which additional scripts? It's only the seed script. One is out and other is in?

Comment (mkiterian) [reply package.json]: `downdev` and `migratedev`

Comment (placideirandora) [reply package.json]: I use them for creating and removing tables without PG ADMIN.

File: src/controllers/articleController.js (hunk 4 above)
Comment (placideirandora) [reply src/controllers/articleController.js]: Okay

File: package.json (hunk 1 above)
Comment (placideirandora) [reply package.json]: Okay. I will remove them!

**Pull request merged**